### ISSUE TITLE
Creating multiple projects from one module

### ISSUE THREAD

**Issue by MarioUhrik** [terraform-google-modules/terraform-google-project-factory#157]
Hello,
my intention is to create a script **script.sh** that uses one Terraform config **main.tf** with a single project factory module in order to create an arbitrary number of GCP projects.
The idea of the usage is:
```
./script.sh project-x
./script.sh project-y
./script.sh project-z
```
where projects x,y,z are GCP projects that i'd like to create and **script.sh** does roughly
`terraform apply -auto-approve -var "credentials_path=credentials.json" -var "project_name=${1}"`
and the **main.tf** is only composed of a single module which obtains a different project name from **script.sh** each time.

The problem is that if the first call to create _project-x_ succeeds, a Terraform state file is created, and subsequent script call to create _project-y_ would react to the existing Terraform state file by deleting _project-x_. 
What I'd like to achieve is to somehow ensure that the sequence of the **script.sh** calls leads to creation and peaceful co-existence of all projects.

What would you recommend, please ?

IDEAS:
- I am experimenting with **script.sh** doing the `terraform apply` calls in separate directories specific to the project name, thus getting Terraform state files only related to specific projects. E.g. `./script.sh project-y` would create directory `project-y` and do the rest of the work inside of it. This seems to kinda work, but I am not sure whether I like this solution.

- Another idea I have is to maintain a dynamic Terraform config file **main.tf** where `./script.sh project-y`would add a module for project _project-y_ into **main.tf**, then do `terraform apply` and hopefully let Terraform maintain a single state file for all the projects without problems.

**Comment by morgante** [comment]
@MarioUhrik Did you solve this?

**Comment by MarioUhrik** [comment]
I have tried both approaches personally, and both worked as expected.
I decided to keep using the first approach for now.

This issue isn't really a problem within terraform-google-project-factory, but more like a discussion or asking for advice. There didn't seem to be any interest in this topic from other people, and I have moved on since then. Hence I decided to close the issue, but we can reopen if you would prefer. Sorry for any possible inconvenience I might have caused, I intended none.
